xmpp.go

```diff
@@ -238,10 +238,9 @@ func (o Options) NewClient() (*Client, error) {
 		if o.TLSConfig != nil {
 			tlsconn = tls.Client(c, o.TLSConfig)
 		} else {
-			DefaultConfig.ServerName = host
```

Comment:
This line of code confused me a bit. The server name is set in DefaultConfig, then it is copied, then server name is set again in the copy?

I am also not sure why we would want to set the server name on the global DefaultConfig.